### PR TITLE
chore(rustc-perf): update aws terraform provider from 3.59 to 4.67

### PR DESCRIPTION
**Pull request by MarcoIeni**

### CODE REVIEW

File: terraform/rustc-perf/s3.tf [hunk 1]
```diff
@@ -7,6 +7,7 @@ module "static_website" {
   domain_name            = "perf-data.rust-lang.org"
   origin_domain_name     = aws_s3_bucket.bucket.bucket_regional_domain_name
   origin_access_identity = aws_cloudfront_origin_access_identity.bucket.cloudfront_access_identity_path
+  response_policy_id     = todo
```
[the remark this answers is not in the document]
Comment (MarcoIeni) [reply terraform/rustc-perf/s3.tf:10]: We need to create a new resource.

**Review by jdno — COMMENTED** [review]
As discussed last week, let's try to create a [`aws_cloudfront_response_headers_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_response_headers_policy) and see what diff Terraform will produce. 🙂

**Comment by MarcoIeni** [comment]
plan:

```
Terraform will perform the following actions:

  # aws_cloudfront_response_headers_policy.response_policy will be created
  + resource "aws_cloudfront_response_headers_policy" "response_policy" {
      + comment = "Policy for RustcPerf website"
      + etag    = (known after apply)
      + id      = (known after apply)
      + name    = "RustcPerfPolicy"

      + security_headers_config {
          + content_type_options {
              + override = true
            }
          + frame_options {
              + frame_option = "DENY"
              + override     = true
            }
          + referrer_policy {
              + override        = true
              + referrer_policy = "no-referrer"
            }
          + strict_transport_security {
              + access_control_max_age_sec = 63072000
              + override                   = true
            }
          + xss_protection {
              + mode_block = true
              + override   = true
              + protection = true
            }
        }
    }

  # module.static_website.aws_cloudfront_distribution.website will be updated in-place
  ~ resource "aws_cloudfront_distribution" "website" {
      ~ http_version                   = "http2" -> "http2and3"
        id                             = "E355NZKQL79WTL"
        tags                           = {}
        # (19 unchanged attributes hidden)

      ~ default_cache_behavior {
          + cache_policy_id            = "658327ea-f89d-4fab-a63d-7e88639e58f6"
          + response_headers_policy_id = (known after apply)
            # (11 unchanged attributes hidden)

          - forwarded_values {
              - headers                 = [] -> null
              - query_string            = false -> null
              - query_string_cache_keys = [] -> null

              - cookies {
                  - forward           = "none" -> null
                  - whitelisted_names = [] -> null
                }
            }
        }

      ~ viewer_certificate {
          ~ minimum_protocol_version       = "TLSv1" -> "TLSv1.2_2021"
            # (3 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 1 to add, 1 to change, 0 to destroy.
╷
│ Warning: Argument is deprecated
│
│   with aws_s3_bucket.bucket,
│   on s3.tf line 90, in resource "aws_s3_bucket" "bucket":
│   90: resource "aws_s3_bucket" "bucket" {
│
│ Use the aws_s3_bucket_versioning resource instead
│
│ (and 5 more similar warnings elsewhere)
╵
```

**Comment by MarcoIeni** [comment]
@Kobzol does the policy I added look reasonable?
How can I make sure nothing breaks after I deploy this change?

I would like to understand and document how `perf-data.rust-lang.org` is used 👍

**Comment by Kobzol** [comment]
Wait, `perf-data.rust-lang.org`? I actually have no idea what that is. I thought that you're talking about `perf.rust-lang.org` :)

**Comment by MarcoIeni** [comment]
it's used in rustc-perf: https://github.com/search?q=repo%3Arust-lang%2Frustc-perf%20perf-data.rust-lang.org&type=code

**Comment by MarcoIeni** [comment]
Ok, thanks! I'll merge and apply this PR. hopefully there will be no angry users 😁

**Comment by MarcoIeni** [comment]
I just applied. Let me know if there's any issue

**Pull request merged**